CMakeLists.txt

```diff
@@ -36,20 +37,21 @@ else()
   set(extra-libs "")
 endif()
 
-try_run(TEST_RUN_RESULT
-  TEST_COMPILE_RESULT
-  ${CMAKE_CURRENT_BINARY_DIR}/
-  ${PROJECT_SOURCE_DIR}/test/cpp17/fs.cpp
-  LINK_LIBRARIES ${extra-libs}
-  OUTPUT_VARIABLE var)
+if(TEST_C++17_COMPILER)
```

Comment:
I would rather not have this as an option, since I did encounter a few segfaults in older versions of `gcc` because of features not yet complete, although exposed in the `filesystem` library.

It's more reasonable to disable the test for known correct compilers. For instance:
`clang >= 10.0.0` and `gcc >= 9.0.0`.